pkg/cloud/services/ec2/securitygroups.go

```diff
@@ -458,6 +459,23 @@ func (s *Service) getSecurityGroupIngressRules(role infrav1.SecurityGroupRole) (
 				},
 			},
 		}, nil
+	case infrav1.SecurityGroupAPIServerLB:
+		return infrav1.IngressRules{
+			{
+				Description: "Kubernetes API",
+				Protocol:    infrav1.SecurityGroupProtocolTCP,
+				FromPort:    int64(s.scope.APIServerPort()),
+				ToPort:      int64(s.scope.APIServerPort()),
+				CidrBlocks:  []string{anyIPv4CidrBlock},
+			},
+			{
+				Description: "Path MTU Discovery",
+				Protocol:    infrav1.SecurityGroupProtocolICMP,
+				FromPort:    3,
+				ToPort:      4,
+				CidrBlocks:  []string{anyIPv4CidrBlock},
+			},
```

Comment:
This makes me wonder if we should have a rule to allow PMTU for the other security groups (other than `SecurityGroupLB`) as well.

Reply:
I think you are correct @detiber  and adding PTMU in other security groups starts to widen the scope of this PR. I've removed PMTU for now and will include that in a separate PR.